### PR TITLE
[libmediainfo] update to 24.01

### PR DESCRIPTION
**Pull request by jiayuehua**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [ ] ~Any patches that are no longer applied are deleted from the port's directory.~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/libmediainfo/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "libmediainfo",
-  "version": "23.10",
+  "version-string": "24.01",
```
Comment (JonLiu1993) [inline ports/libmediainfo/vcpkg.json:3]:
```suggestion
  "version": "24.01",
```

Comment (JonLiu1993) [reply ports/libmediainfo/vcpkg.json:3]: @jiayuehua, thanks for the PR, we don't recommend using "version string", especially when "version" is available

**Comment by JonLiu1993** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/libmediainfo/vcpkg.json (hunk 1 above)
Comment (dg0yt) [reply ports/libmediainfo/vcpkg.json:3]: Unfortunately, we cannot use leading zeros with scheme `version`.
What can be done: Use `"version": "24.1"`, and use a REGEX REPLACE to generate other formats in the portfile.

**Review by JonLiu1993 — DISMISSED** [review]

File: ports/libmediainfo/portfile.cmake [hunk 2]
```diff
@@ -1,7 +1,16 @@
+string(REGEX REPLACE "^([0-9]*)[.].*" "\\1" MAJOR "${VERSION}")
+string(REGEX REPLACE "^.*[.]([0-9]*)" "\\1" MINOR "${VERSION}")
+string(LENGTH "${MINOR}" minor_version_length )
+
+set (padding "")
+if(minor_version_length EQUAL 1)
+    set (padding "0")
+endif()
```
Comment (dg0yt) [inline ports/libmediainfo/portfile.cmake:8]:
```suggestion
string(REGEX REPLACE "[.]([0-9])\$" ".0\\1" VERSION_STR "${VERSION}")
```

File: ports/libmediainfo/portfile.cmake [hunk 3]
```diff
@@ -1,7 +1,16 @@
...
+
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO MediaArea/MediaInfoLib
-    REF "v${VERSION}"
+    REF "v${MAJOR}.${padding}${MINOR}"
```
Comment (dg0yt) [inline ports/libmediainfo/portfile.cmake:13]:
```suggestion
    REF "v${VERSION_STR}"
```

**Review by dg0yt — CHANGES_REQUESTED** [review]
Simplify.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**